Kernel/CMakeLists.txt

```diff
@@ -495,6 +495,18 @@ add_compile_options(-fno-exceptions)
 # FIXME: remove -nodefaultlibs after the next toolchain update
 add_compile_options(-nodefaultlibs -nostdlib)
 
+# Auto initialize trivial types on the stack, we use "pattern" as 
+# it's the only option portable across compilers going forward.
+#
+# This is designed to help avoid uninitialized variables bugs and
+# information disclosures coming from the kernel stack. 
+#
+# FIXME: It appears to conflict with something during the boot of the
+# aarch64 kernel, we should investigate and remove this special case.
```

Comment:
Regarding the aarch64 stuff, it is crashing here:
https://github.com/SerenityOS/serenity/blob/ef1c97e3d8c73b50649f6c03d500bfb66cb67677/Kernel/Arch/aarch64/Registers.h#L212

This pattern (packed structs) has caused us so many problems already, I think it's time to refactor this code to use shifts etc. But that's not relevant for this PR :+1: